### PR TITLE
Fix #2296 : 2D arrays - assert size of first dimension only

### LINKED ISSUE #2296: 2D arrays: assert size of first dimension only
#### Summary

In assertj 3.17.0, for 2D arrays like `byte[][]` the `hasSize(int)` method has been replaced by `hasDimensions(int, int)`.

It would be useful to keep the `hasSize(int)` as well. The length of the second dimension might be different for each entry, or it might simply be irrelevant to the test.

#### Example

```java
byte[][] bytes = {{1}, {1, 2}, {1, 2, 3}};
assertThat(bytes).hasDimensions(3, /* problem: second dimension is variable length */);
```

### PR DESCRIPTION
**Pull request by SekarMylsamy**
- Fixes #2296
- Unit tests : YES
- Javadoc with a code example (on API only) : Y

### CODE REVIEW

**Review by scordio — COMMENTED** [review]
Thanks @SekarMylsamy, here my feedback.

File: src/main/java/org/assertj/core/api/Byte2DArrayAssert.java [hunk 1]
```diff
@@ -212,6 +212,29 @@ public Byte2DArrayAssert hasDimensions(int expectedFirstDimension, int expectedS
...
+   * assertThat(new byte[][] {{1, 2, 3}, {4, 5, 6}}).hasRowSize(3);
+   * assertThat(new byte[][] {{1, 2, 3}, {4, 5, 6, 7}}).hasRowSize(1); </code></pre>
+   *
+   * @param expectedFirstDimension the expected number of values in first dimension of the actual {@code byte[][].
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual number of rows are not equal to the given one.
+   */
+  @Override
+  public Byte2DArrayAssert hasRowSize(int expectedFirstDimension) {
+    byte2dArrays.assertHasDimensions(info, actual, expectedFirstDimension, 0);
```
[the remark this answers is not in the document]
Comment (SekarMylsamy) [reply src/main/java/org/assertj/core/api/Byte2DArrayAssert.java:233]: Sure @scordio. I will update the implementation. Regarding the naming, will wait for alignment from everyone.

[the remark this answers is not in the document]
Comment (scordio) [reply src/main/java/org/assertj/core/api/Byte2DArrayAssert.java:233]: As discussed in #2296, let's go ahead with `hasNumberOfRows`.

[the remark this answers is not in the document]
Comment (SekarMylsamy) [reply src/main/java/org/assertj/core/api/Byte2DArrayAssert.java:233]: Thanks, will work on it.

**Comment by WojciechZankowski** [comment]
Hi @scordio, do you think I could take over this issue and finalize it in a new PR?

**Comment by scordio** [comment]
@SekarMylsamy are you okay with @WojciechZankowski taking over this change?

**Comment by SekarMylsamy** [comment]
> @SekarMylsamy are you okay with @WojciechZankowski taking over this change?

Sure, @WojciechZankowski please take over.

**Comment by scordio** [comment]
Thanks for the feedback, @SekarMylsamy!

**Comment by scordio** [comment]
Superseded by #2461.

**Pull request closed without merging**